### PR TITLE
Slightly refactor store request packet call-tree

### PR DESCRIPTION
**Pull request by drichmond**
Packets should use SW, but this required passing sz to the function that set the op. This seemed to break an abstraction so I merged hb_mc_manycore_format_store_request_packet into its single callsite and set SW/STORE there.

Also, refactored to use HB_MC_SUCCESS instead of 0.

### CODE REVIEW

**Comment by drichmond** [comment]
(I have not run regression on this yet, but will post when I have)

File: libraries/bsg_manycore.cpp [hunk 1]
```diff
@@ -445,13 +431,13 @@ static int hb_mc_manycore_format_load_request_packet(hb_mc_manycore_t *mc,
 {
         int r;
 
-        if ((r = hb_mc_manycore_format_request_packet(mc, pkt, npa)) != 0)
+        if ((r = hb_mc_manycore_format_request_packet(mc, pkt, npa)) HB_MC_SUCCESS)
```
Comment (dpetrisko) [inline libraries/bsg_manycore.cpp:434]: dropped an != ?

File: libraries/bsg_manycore.cpp [hunk 2]
```diff
@@ -461,13 +447,13 @@ static int hb_mc_manycore_format_cache_op_request_packet(hb_mc_manycore_t *mc,
...
```
Comment (dpetrisko) [inline libraries/bsg_manycore.cpp:450]: !=

Comment (drichmond) [reply libraries/bsg_manycore.cpp:450]: HA. Shows I didn't even run compilation. Yes, I screwed something up in my rebase.

**Pull request merged**